src/coreclr/nativeaot/Runtime/threadstore.cpp

```diff
@@ -31,6 +31,9 @@ volatile uint32_t RhpTrapThreads = (uint32_t)TrapThreadsFlags::None;
 
 GVAL_IMPL_INIT(PTR_Thread, RhpSuspendingThread, 0);
 
+GPTR_DECL(Thread, g_pFinalizerThread);
+EXTERN_C NATIVEAOT_API void REDHAWK_CALLCONV InitInlineThreadStatics(void* pTypeManager);
```

Comment:
This should use the GetClasslibFunction mechanism. We don't assume a single corelib in the rest of the runtime code.

Reply:
ClassLib is the library that defines System.Object in the runtime lingo. There can be multiple of them. If we're calling into something that is defined in System.Private.CoreLib, it needs to use GetClassLibFunction because:

1. There can be multiple of those.
2. We should ideally keep the option to compile src/coreclr/nativeaot/runtime + src/coreclr/nativeaot/Runtime.Base into a separate DLL like we had in .NET Native. If we have a hard reference to a symbol from CoreLib, this would have unresolved imports.

I just realized it's a pre-existing problem but it only jumped out now because the API is not prefixed `Rhp`. Apis prefixed `Rhp` should be in Runtime.Base, not in CoreLib. This one should be prefixed Rhp. We can move it to Runtime.Base later.

Reply:
I see. What I hear is that threadstatic support should live in the Runtime.Base, but for now we shoudl prefix the helper `Rhp`. 

Do we have the same issue with regular statics and constructor runner?

Reply:
> Do we have the same issue with regular statics and constructor runner?

No, those are meant to be provided by CoreLib. The compiler is responsible for choosing the cctor runner during codegen and it looks for it in CoreLib. If it can't find any, it emits code to initialize all statics before Main and considers all types preinitialized. The runtime doesn't care about the cctor runner.

On the other hand, threadstatics are a runtime concern because we attach them to the thread. So it needs to be in Runtime.Base/native runtime.

Reply:
> threadstatics are a runtime concern because we attach them to the thread. 

Strictly speaking, threadstatics do not have to be a runtime concern. The storage for them can be created and managed by the compiler and the app CoreLib. The only thing that the runtime needs to know is how to report them to the GC. The reporting can be done via GC handle, but there are some advantages for reporting them via a special runtime-provided mechanism.

Reply:
This is resolved, since we no longer initialize on thread attach. This part was reverted back to original - "on first use" pattern.